networkx/algorithms/d_separation.py

```diff
@@ -123,7 +123,44 @@
 import networkx as nx
 from networkx.utils import UnionFind, not_implemented_for
 
-__all__ = ["d_separated", "minimal_d_separator", "is_minimal_d_separator"]
+__all__ = ["d_separated", "minimal_d_separated", "find_minimal_d_separator"]
+
+
+def ancestors(G, start_nodes):
+    """Computes the ancestors of a set of nodes in a graph with directed edges.
+
+    By definition, the anterior set includes the start nodes.
+
+    Parameters
+    ----------
+    G : nx.DiGraph
+        The graph.
+    start_nodes : set
+        Set of nodes which are always included in the found separating set.
+
+    Returns
+    -------
+    visited : set
+        The anterior set of nodes
+
+    References
+    ----------
+    .. [1] B. van der Zander, M. Liśkiewicz, and J. Textor, “Separators and Adjustment
+       Sets in Causal Graphs: Complete Criteria and an Algorithmic Framework,” Artificial
+       Intelligence, vol. 270, pp. 1–40, May 2019, doi: 10.1016/j.artint.2018.12.006.
+    """
+
+    queue = deque(start_nodes)
+    visited = set()
+
+    while queue:
+        m = queue.popleft()
+        for x, _ in G.in_edges(nbunch=m):
+            if x not in visited:
+                queue.append(x)
+                visited.add(x)
+
+    return visited.union(start_nodes)
```

Comment:
We can use the existing networkx implementation: https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.dag.ancestors.html

Reply:
Seems like the existing implementation doesn't support taking ancestors of a set?

Reply:
I've modified it accordingly nonetheless